nemo/collections/nlp/modules/common/transformer/transformer.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+from omegaconf.omegaconf import MISSING
+
+from nemo.collections.nlp.modules.common.decoder_module import DecoderModule
+from nemo.collections.nlp.modules.common.encoder_module import EncoderModule
+from nemo.collections.nlp.modules.common.transformer.transformer_decoders import TransformerDecoder
+from nemo.collections.nlp.modules.common.transformer.transformer_encoders import TransformerEncoder
+from nemo.collections.nlp.modules.common.transformer.transformer_modules import TransformerEmbedding
+
+
+@dataclass
+class TransformerConfig:
+    hidden_size: int = MISSING
+    num_layers: int = MISSING
+    inner_size: int = MISSING
+
+    # embedding
+    max_sequence_length: int = 512
+    num_token_types: int = 2
+    embedding_dropout: float = 0.0
+    learn_positional_encodings: bool = False
+
+    # transformer
+    num_attention_heads: int = 1
```

Comment:
this is very important parameter which is related with hidden_size, let's keep it MISSING like hidden_size and not provide a default value for it.

Reply:
I think it should be set to `MISSING`. Will update the configs.